wallets/index.js

```diff
@@ -1,452 +1,237 @@
-  const { me } = useMe()
-
-  const { data, refetch: refetchConfig } = useQuery(WALLET_BY_TYPE, { variables: { type: wallet?.walletType }, skip: !wallet?.walletType })
-
-  const walletId = data?.walletByType?.id
-  const serverConfig = {
-    id: walletId,
-    priority: data?.walletByType?.priority,
-    enabled: data?.walletByType?.enabled,
-    ...data?.walletByType?.wallet
-  }
-  delete serverConfig.__typename
-
-  const autowithdrawSettings = autowithdrawInitial({ me })
-  const config = { ...serverConfig, ...autowithdrawSettings }
-
-  const saveConfig = useCallback(async ({
-    autoWithdrawThreshold,
-    autoWithdrawMaxFeePercent,
-    autoWithdrawMaxFeeTotal,
-    priority,
-    enabled,
-    ...config
-  }, { priorityOnly }) => {
-    try {
-      const mutation = generateMutation(wallet)
-      return await client.mutate({
-        mutation,
-        variables: {
-          ...config,
-          id: walletId,
-          settings: {
-            autoWithdrawThreshold: Number(autoWithdrawThreshold),
-            autoWithdrawMaxFeePercent: Number(autoWithdrawMaxFeePercent),
-            autoWithdrawMaxFeeTotal: Number(autoWithdrawMaxFeeTotal),
-            priority,
-            enabled
-          },
-          priorityOnly
+  }, [me?.privates?.autoWithdrawMaxFeePercent, me?.privates?.autoWithdrawThreshold, me?.privates?.autoWithdrawMaxFeeTotal])
+
+  // whenever the vault key is set, and we have local wallets,
+  // we'll send any merged local wallets to the server, and delete them from local storage
+  const syncLocalWallets = useCallback(async encrypt => {
+    const walletsToSync = wallets.filter(w =>
+      // only sync wallets that have a local config
+      localWallets.some(localWallet => localWallet.def.name === w.def.name && !!localWallet.config)
+    )
+    if (encrypt && walletsToSync.length > 0) {
+      for (const wallet of walletsToSync) {
+        const mutation = generateMutation(wallet.def)
+        const append = {}
+        // if the wallet has server-only fields set, add the settings to the mutation variables
+        if (wallet.def.fields.some(f => f.serverOnly && wallet.config[f.name])) {
+          append.settings = settings
         }
-      })
-    } finally {
-      client.refetchQueries({ include: ['WalletLogs'] })
-      refetchConfig()
+        const variables = await upsertWalletVariables(wallet, encrypt, append)
+        await client.mutate({ mutation, variables })
+      }
+      removeLocalWallets()
     }
-  }, [client, walletId])
-
-  const clearConfig = useCallback(async () => {
-    // only remove wallet if there is a wallet to remove
-    if (!walletId) return
+  }, [wallets, localWallets, removeLocalWallets, settings])
 
-    try {
-      await client.mutate({
-        mutation: REMOVE_WALLET,
-        variables: { id: walletId }
-      })
-    } finally {
-      client.refetchQueries({ include: ['WalletLogs'] })
-      refetchConfig()
+  const unsyncLocalWallets = useCallback(() => {
```

Comment:
this is called when device sync is intentionally disconnected ... this way it preserves the config, ie reset does not necessarily result in data loss.